src/helpers/CountriesDataContext.jsx

```diff
@@ -0,0 +1,28 @@
+import { createContext, useState } from 'react';
+import { API_URL } from '../config';
+
+export const CountriesContext = createContext();
+
+export function CountriesDataContext(props) {
+  const [countries, setCountries] = useState([]);
+
+  const fetchDate = async () => {
+    if (countries.length === 0) {
+      fetch(API_URL)
+        .then((response) => response.json())
+        .then((data) => {
+          setCountries(data);
+          console.log(data);
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  };
+
+  const value = {
```

Comment:
Wrap this one into useMemo and then check number of renders on the home page. If necessary also wrap fetchDate into useCallback.